From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Fenix - Secret Settings visibility

Provides support for controlling the visibility of UI settings within the Secret Settings fragment.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 7cf7648c813a..ee18c45c9fad 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -4,6 +4,7 @@
 
 package org.mozilla.fenix.settings
 
+import org.ironfoxoss.ironfox.utils.FenixSettingsVisibilityManager
 import android.os.Build
 import android.os.Bundle
 import android.view.LayoutInflater
@@ -562,6 +563,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFra
             isChecked = context.settings().longfoxEnabled
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
+        FenixSettingsVisibilityManager.SecretSettingsFragment(requireContext(), this)
     }
 
     override fun onDisplayPreferenceDialog(preference: Preference) {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 83e0f40483b7..edf076d5ec43 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2122,7 +2122,7 @@ class Settings(
      */
     private fun isAddressFeatureEnabled(context: Context): Boolean {
         val locale = LocaleManager.getCurrentLocale(context) ?: LocaleManager.getSystemDefault()
-        val debugRepository = if (Config.channel.isNightlyOrDebug) {
+        val debugRepository = if (true) {
             SharedPrefsAddressesDebugRegionRepository(context)
         } else {
             EmptyAddressesDebugRegionRepository()
-- 
